Tests/Common/SimulatedBasicsStation.cs

```diff
@@ -0,0 +1,173 @@
+                });
+                await routerWebsocketClient.Start();
+                using var ms = new MemoryStream();
+                using var writer = new Utf8JsonWriter(ms);
+
+                await routerWebsocketClient.SendInstant(JsonSerializer.Serialize(new
+                {
+                    router = this.stationEUI
+                }));
+
+                await Task.Delay(5000);
+
+                // We could take here additional information e.g. the basicsstation region or the address.
+                Assert.Single(routerReceivedMessage);
+                await routerWebsocketClient.Stop(WebSocketCloseStatus.NormalClosure, "closing WS");
+                var factory = new Func<ClientWebSocket>(() => new ClientWebSocket
+                {
+                    Options =
+                        {
+                            KeepAliveInterval = TimeSpan.FromSeconds(5),
+                        }
+                });
+
+                DataWebsocketClient = new WebsocketClient(new Uri(LnsUri, $"router-data/{ stationEUI }"), factory)
+                {
+                    ReconnectTimeout = TimeSpan.FromSeconds(5)
+                };
+                DataWebsocketClient.ReconnectionHappened.Subscribe(info =>
+                {
+                    Console.WriteLine("Reconnection happened, type: " + info.Type);
+                });
+                DataWebsocketClient.MessageReceived.Subscribe(msg =>
+                {
+                    routerReceivedMessage.Add(msg.Text);
+                    if (this.subscribers.Count > 0)
+                    {
+                        Func<string, bool> subscriberToRemove = null;
+
+                        foreach (var subscriber in this.subscribers.Where(subscriber => subscriber(msg.Text)))
+                        {
+                            subscriberToRemove = subscriber;
+                            break;
+                        }
+
+                        if (subscriberToRemove != null)
+                        {
+                            this.subscribers.Remove(subscriberToRemove);
+                        }
+                    }
+                    Console.WriteLine("Message received: " + msg);
+                });
+                await DataWebsocketClient.Start();
+
+                var versionMessage = new LnsVersionRequest(stationEUI, "2", "1", "test", 2, "");
+                await DataWebsocketClient.SendInstant(JsonSerializer.Serialize(versionMessage));
+                await Task.Delay(5000);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("ERROR: " + ex);
+            }
+        }
+
+        //// Sends unconfirmed message
+        public async Task SendDataMessageAsync(LoRaRequest LoRaRequest)
+        {
+            var payload = (LoRaPayloadData)LoRaRequest.Payload;
+
+            // catch me if you can.
+            payload.DevAddr.Span.Reverse();
+
+            var msg = JsonSerializer.Serialize(new
```

Comment:
It would be great if we could run the simulator tests from now on (or at least part of them) to ensure that we don't break such logic.

Reply:
Agree - I added a work item for this. I think the tests are not yet in shape to bring enough values . #1223